diff --git a/node_modules/ra-data-odata-server/src/index.ts b/node_modules/ra-data-odata-server/src/index.ts
index 9238458..7ca4a15 100644
--- a/node_modules/ra-data-odata-server/src/index.ts
+++ b/node_modules/ra-data-odata-server/src/index.ts
@@ -74,7 +74,7 @@ const ra_data_odata_server = async (
       return id;
     }
   }
-
+/*
   return resource_id_mapper(
     {
       getResources: () => Object.values(resources).map(r => r.Name),
@@ -100,7 +100,7 @@ const ra_data_odata_server = async (
 
         for (const filterName in params.filter) {
           o = o.filter(
-            `Contains(${filterName},'${params.filter[filterName]}')`
+            //`Contains(${filterName},'${params.filter[filterName]}')`
           );
         }
 
